### PR TITLE
 Moves to freestyle.free

### LINKED ISSUE #469: Move `@free` and `@module` and others under the `freestyle.free` namespace
```
freestyle.@free -> freestyle.free.@free
freestyle.@module -> freestyle.free.@module
package object freestyle -> package freestyle ; package object free
freestyle.implicits -> freestyle.free.implicits
```
The idea of this Pr is to properly namespace `@free` dependent code under the `free` package so we have parity with `freestyle.tagless`.
Additionally all modules and algebras that target only free monads should be namespaced below that as well so we can create counterparts with the same semantics in the `tagless` namespace.

### PR DESCRIPTION
**Pull request by AdrianRaFo**
Fixes #469

### CODE REVIEW

**Review by raulraja — APPROVED** [review]
:clap:

**Comment by raulraja** [comment]
@juanpedromoreno given the only module failing is the docs and this requires a multi step release, are we ready to merge this and release all other modules?

**Comment by juanpedromoreno** [comment]
Deployment is not going to work, we would need to comment the part of the pipeline where docs are being checked.

**Review by juanpedromoreno — APPROVED** [review]

File: modules/async/async/shared/src/main/scala/free/async.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/*
```
Comment (juanpedromoreno) [inline modules/async/async/shared/src/main/scala/free/async.scala:1]: @raulraja @AdrianRaFo Now we have like two `async.scala` files, one inside the free package and another one outside. Is this the expected behavior?

**Pull request merged**